### PR TITLE
Add support for Windows timeouts for comm handles

### PR DESCRIPTION
**Pull request by jcreekmore**

### CODE REVIEW

File: src/reader.rs [hunk 1]
```diff
@@ -27,33 +33,50 @@ use super::utils;
...
 where
-    H: Read + AsRawFd + Seek,
+    H: Seek,
 {
     fn seek(&mut self, pos: SeekFrom) -> Result<u64> {
         self.handle.seek(pos)
     }
 }
 
+#[cfg(unix)]
```
Comment (jonhoo) [inline src/reader.rs:79]: Maybe forward `AsRawHandle` for `cfg(windows)` too?

File: src/writer.rs [hunk 2]
```diff
@@ -50,15 +54,44 @@ where
...
 where
-    H: Write + AsRawFd + Seek,
+    H: Seek,
 {
     fn seek(&mut self, pos: SeekFrom) -> Result<u64> {
         self.handle.seek(pos)
     }
 }
 
+#[cfg(unix)]
```
Comment (jonhoo) [inline src/writer.rs:94]: Same point re forwarding on windows here.

**Comment by jonhoo** [comment]
This looks reasonable to me! Pretty much exactly what I imagined.

**Review by jonhoo — APPROVED** [review]

File: src/reader.rs (hunk 1 above)
Comment (jcreekmore) [reply src/reader.rs:79]: I mean, I _could_, but I was really only doing that because the helper function I used for unix used it behind the scenes.

**Comment by jcreekmore** [comment]
The Windows bit is a bit of a mess (see the discussion on #14), so I am closing this for now.

**Pull request closed without merging**